src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/InstructionDecoderTests.cs

```diff
@@ -402,6 +402,144 @@ class C
             Assert.Equal("System.Action<System.Func<object>>", GetReturnTypeName(source, "C.M1", [typeof(object)]));
         }
 
+        [Fact]
+        public void GetCompactName_Members()
+        {
+            var source = """
+                using System;
+                namespace System.Runtime.CompilerServices
+                {
+                    public class IsExternalInit { }
+                }
+                class C
+                {
+                    static C() { }
+                    C(int x) { }
+                    ~C() { }
+                    object F() => null;
+                    object P1 { get; }
+                    object P2 { set { } }
+                    object P3 { init { } }
+                    object this[int i] { get { return null; } set { } }
+                    event EventHandler E;
+                    public static C operator+(C c) => c;
+                    public static implicit operator int(C c) => 0;
+                    public static explicit operator string(C c) => "";
+                }
+                static class E
+                {
+                    static void M(this string x) { }
+                }
+                """;
+            var compilation = CreateCompilation(source);
+            var containingType = compilation.GlobalNamespace.GetTypeMember("C");
+            VerifyMethodName(compilation, containingType.GetMethod("F"), "C.F()", "F");
+            VerifyMethodName(compilation, containingType.GetMethod("get_P1"), "C.P1.get()", "P1");
+            VerifyMethodName(compilation, containingType.GetMethod("set_P2"), "C.P2.set(value)", "P2");
+            VerifyMethodName(compilation, containingType.GetMethod("set_P3"), "C.P3.init(value)", "P3");
+            VerifyMethodName(compilation, containingType.GetMethod("get_Item"), "C.this[int].get(i)", "this");
+            VerifyMethodName(compilation, containingType.GetMethod("set_Item"), "C.this[int].set(i, value)", "this");
+            VerifyMethodName(compilation, containingType.GetMethod("add_E"), "C.E.add(value)", "E");
+            VerifyMethodName(compilation, containingType.GetMethod("remove_E"), "C.E.remove(value)", "E");
+            VerifyMethodName(compilation, compilation.GlobalNamespace.GetTypeMember("E").GetMethod("M"), "E.M(x)", "M");
+            VerifyMethodName(compilation, containingType.GetMethod(".cctor"), "C.C()", "C");
+            VerifyMethodName(compilation, containingType.GetMethod(".ctor"), "C.C(x)", "C");
+            VerifyMethodName(compilation, containingType.GetMethod("Finalize"), "C.~C()", "~C");
+            VerifyMethodName(compilation, containingType.GetMethod("op_UnaryPlus"), "C.operator +(c)", "operator +");
+            VerifyMethodName(compilation, containingType.GetMethod("op_Implicit"), "C.implicit operator int(c)", "implicit operator int");
```

Comment:
>implicit operator int [](http://example.com/codeflow?start=114&length=21)

Since you can't define both an implicit and an explicit overload of the same operator in C#, could we spare the `implicit`/`explicit` modifier?
Also, consider testing with `checked` user-defined operator (that one does come in pairs, so we may want to include `checked` modifier)

Reply:
> Also, consider testing with `checked` user-defined operator (that one does come in pairs, so we may want to include `checked` modifier)

I don't think we need to include modifiers to disambiguate one method from another. We're not including parameters for example, so overloaded methods will be ambiguous.

Reply:
> Since you can't define both an implicit and an explicit overload of the same operator in C#, could we spare the `implicit`/`explicit` modifier?

Removing `implicit` and `explicit` would require modifying the set of tokens generated by `SymbolDisplay`. For now, let's keep the modifier.